ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -202,6 +202,24 @@
   public static final String KSQL_AUTH_CACHE_MAX_ENTRIES_DOC = "Controls the size of the cache "
       + "to a maximum number of KSQL authorization responses entries.";
 
+  public static final String KSQL_INTERNAL_HIDDEN_TOPICS_CONFIG = "ksql.internal.hidden.topics";
+  public static final String KSQL_INTERNAL_HIDDEN_TOPICS_DEFAULT = "_confluent.*,__confluent.*"
+      + ",_schemas,__consumer_offsets,__transaction_state,connect-configs,connect-offsets,"
+      + "connect-status,connect-statuses";
+  public static final String KSQL_INTERNAL_HIDDEN_TOPICS_DOC = "List of topics that will not be "
+      + "visible when running the SHOW TOPICS command unless SHOW ALL TOPICS is used. This list "
+      + "is comma separated and may use Java regular expressions to specify each topic (i.e. "
+      + " _confluent.* accepts any topic that starts with the _confluent prefix).";
```

Comment:
Great docs :) 

We could even give them a link to docs on Java regex, for those not familiar, and add details of defaults and topics that KSQL always hidden.

A slightly reworded example is below.

```
Comma-separated list of topic that will be hidden. Entries in the list may be literal topic names or [Java regular expressions](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html). For example, `_confluent.*` will match any topic whose name starts with the `_confluent`).

Hidden topics will not visible when running the `SHOW TOPICS` command unless `SHOW ALL TOPICS` is used. 

The default value hides known system topics from Kafka and Confluent products.

KSQL also marks its own internal topics as hidden. This is not controlled by this config. 
```